Topics/Development_Process/Azure_webapp_deployment_with_nodejs.md

```diff
@@ -1,10 +1,191 @@
+
+## Deployment
+
+Azure has a variety of ways to host your application, including Azure App Service, Azure Functions, Azure Container Instances, Azure Kubernetes Service, and more, which you can choose to suit your specific usecase. 
+
+See more about the different kinds of hosting options and reasons to choose one over the other [here](https://learn.microsoft.com/en-us/azure/developer/intro/hosting-apps-on-azure).
+
+The following will detail how one would deploy a Node.js Web App using Azure App service, and integrate it with Github Actions for CI/CD.
+
+### Azure App Service
+
 Azure application service allows you to build and host web applications, mobile backends and RESTful APL in the selected programming language without managing the infrastructure. It provides automatic scaling and high availability, supports Windows and Linux, and supports automatic deployment from GitHub, Azure Devops or any Git repository. You can refer to our quick start, tutorials and examples to learn how to use Azure application services.
 
-## Depoly Web app on Azure
+#### Create Web app on Azure
+
+From Microsoft's official documentation, here is a quick tutorial to get an example Node.js project deployed:
 
 [deploy web app with node js on Azure](https://learn.microsoft.com/en-us/azure/app-service/quickstart-nodejs?tabs=linux&pivots=development-environment-vscode)
+
+Alternatively, and possibly easier, you can make a very basic web app with Node.js and Express by doing the following:
+
+Note: Learn more about Node.js and Express apps [here](../Tech_Stacks/Express.md).
+
+1. Create a project directory for the Node.js web app
+2. In the root of your project directory, run `npm init` in the terminal (you should have npm and Node.js installed).
+3. In the terminal, run `npm install express`.
+4. Write the following in the directory in a new file, `server.js` (or whatever name you entered as the entry point during `npm init`).
+    ```javascript
+        const express = require("express");
+
+        const app = express();
+
+        const PORT = 3000
+
+        app.get("/", (req, res) => {
+            res.send(`
+            <h1>CSC301 A2 Demo successfully deployed!</h1>
+            <h2>You did it!</h2>
+            `)
+        });
+
+        // Set up server to listen on port 3000
+        app.listen(PORT, () => {
+            console.log(`Listening on port ${PORT}`);
+        });
+    ```
+
+Note that the file structure inside your project directory should currently look something like this:
+
+```
+├── node_modules
+├── package.json
+├── package-lock.json 
+├── server.js 
+└── .gitignore
+```
+
+Have this code in a Github repository to later integrate with Github actions.
+
+Now, you have two options:
```

Comment:
Expand on pros and cons on the two options. Why choose one over other? Or is there no difference really in the end?

Reply:
I do have a brief explanation at each option explaining the main difference between the two methods, but you're right that it makes sense to include the explanation where you have specified, since that's where they'll be making the decision. Thanks!